mne/decoding/base.py

```diff
@@ -0,0 +1,602 @@
+        Save the coeffiscient in the attribute filters_ and
+        computes the attribute patterns_ using [1].
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_features)
+            The data to estimate the coeffiscient.
+        y : array, shape (n_epochs,)
+            The class for each epoch.
+
+        Returns
+        -------
+        self : instance of LinearModel
+            Returns the modified instance.
+
+        References
+        ----------
+        """
+        # fit the Model
+        self.model.fit(X, y)
+        # computes the patterns
+        assert hasattr(self.model, 'coef_'), \
+            "model needs a coef_ attribute to compute the patterns"
+        self.patterns_ = np.dot(X.T, np.dot(X, self.model.coef_.T))
+        self.filters_ = self.model.coef_
+
+        return self
+
+    def transform(self, X, y=None):
+        """Transform the data using the linear model.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_features)
+            The data to transform.
+        y : array, shape (n_epochs,)
+            The class for each epoch.
+
+        Returns
+        -------
+        y_pred : array, shape (n_epochs,)
+            Predicted class label per epoch.
+
+        """
+        return self.model.transform(X)
+
+    def fit_transform(self, X, y):
+        """fit the data and transform it using the linear model.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_features)
+            The data to transform.
+        y : array, shape (n_epochs,)
+            The class for each epoch.
+
+        Returns
+        -------
+        y_pred : array, shape (n_epochs,)
+            Predicted class label per epoch.
+
+        """
+        return self.fit(X, y).transform(X)
+
+    def predict(self, X):
+        """Computes prediction of X.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_features)
+            The data used to compute prediction.
+
+        Returns
+        -------
+        y_pred : array, shape (n_epochs,)
+            The predictions.
+        """
+        return self.model.predict(X)
+
+    def score(self, X, y):
+        """
+        Returns the score of the linear model computed
+        on the given test data.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_features)
+            The data to transform.
+        y : array, shape (n_epochs,)
+            The class for each epoch.
+
+        Returns
+        -------
+        score : float
+            Score of the linear model
+
+        """
+        return self.model.score(X, y)
+
+    def plot_patterns(self, info, times=None, ch_type=None, layout=None,
+                      vmin=None, vmax=None, cmap='RdBu_r', sensors=True,
+                      colorbar=True, scale=None, scale_time=1e3, unit='a.u.',
+                      res=64, size=1, cbar_fmt='%3.1f',
```

Comment:
I added an inverse_transform to the scaler (and to any class in mne.decoding.transformer)
but there's an issue because it reshapes (n_epochs, n_channels, n_times) and patterns are shape (n_channes, n_times). 
I think reshaping with (-1, n_channels, n_times) should fix this issue... I'll try it later and commit if it works.